src/js/tech/tech.js

```diff
@@ -806,39 +839,29 @@ Tech.withSourceHandlers = function(_Tech) {
   _Tech.prototype.setSource = function(source) {
     let sh = _Tech.selectSourceHandler(source, this.options_);
 
+    // Dispose any existing source handler
+    this.disposeSourceHandler();
+    this.off('dispose', this.disposeSourceHandler);
+
     if (!sh) {
       // Fall back to a native source hander when unsupported sources are
       // deliberately set
       if (_Tech.nativeSourceHandler) {
         sh = _Tech.nativeSourceHandler;
       } else {
         log.error('No source hander found for the current source.');
+        return;
```

Comment:
The addition of this `return` and moving of lines 843-844 to above this conditional block are to fix a tiny issue that may have never been manifest in the wild but which I discovered during a test run.

If a Tech has no `nativeSourceHandler` property and `selectSourceHandler` returned `null` because no matching source handlers were found, then the line later on: `this.sourceHandler_ = sh.handleSource(source, this, this.options_);` would throw an exception because `sh` would stay `null`.